scipy/interpolate/_cubic.py

```diff
@@ -392,19 +427,24 @@ class CubicSpline(PPoly):
 
     Notes
     -----
-    Two additional equations are required to determine all coefficients
-    of polynomials on each segment. They are formed from the conditions that
-    at the first and the last interior knots the third derivative is
-    continuous. It essentially means that the first and the second segments
-    (on both ends) are described by the same cubic polynomial. These conditions
-    are called "not-a-knot" and are known to give better interpolation accuracy
-    when nothing is known about end point derivatives [2]_.
+    When a boundary condition is 'not-a-knot' and n = 2, it is replaced by
+    a condition that the first derivative is equal to the linear interpolant
+    slope. When both boundary conditions are 'not-a-knot' and n = 3, the
+    solution is sought as a parabola passing through given points.
+
+    When 'not-a-knot' boundary conditions is applied to both ends, the
+    resulting spline will be the same as returned by `splrep` (with ``s=0``)
+    and `InterpolatedUnivariateSpline`, but those two methods use a
+    representation in B-spline basis.
+
+    Currently `PPoly` can't correctly evaluate periodic polynomials for
+    arbitrary x. To decrease possible confusion, `extrapolate` is set to False
+    by default when ``bc_type='periodic'``. Fow now you can achieve periodic
+    evaluation for any x as follows::
```

Comment:
So that we don't forget: This paragraph will need to be addressed together with gh-6186